HHS/Head-Start-TTADP
# [TTAHUB-2542] Prevent merging of goals on ARs that are in-flight

## Description of change

- Refactor goal similarity merge logic to account for user's flags when querying groups
- Add new column to flag goals that shouldn't be merged unless user has "closed_goal_merge_override" instead of looking at user's permissions when building groups
- Aside from closed curated goals, also prevent merging of goals on AR's that aren't approved (reports that are DRAFT, SUBMITTED, or NEEDS ACTION), using that new flag

## How to test

Goals that have never been on an approved report will not appear on similarity groups, so to test you need to add a goal that's already been an approved report to a draft report:

1. Create goal similar to "AAA 1" by adding it to an AR and approving it. 
2. Create goal similar to "AAA 2"  by adding it to an AR and approving it. 
3. Visit RTTAPA page. You are prompted to merge the two goals
4. Create a new AR and add "AAA 1". The AR is draft.
5. You need to impersonate a user without the "closed_merge_goal" feature flag (and also not an admin). Visit RTTAPA page. You are NOT prompted to merge the two goals.
6. Approve AR with "AAA 1".
7. Visit RTTAPA page. You are prompted to merge the two goals

Repeat the test, but on step 5 view the RTR as a user with the closed_merge_goal feature flag/admin access. The goal merge will appear as an option.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2542

## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
